spring-cloud-data-module-deployers/spring-cloud-data-module-deployer-cloudfoundry/src/main/java/org/springframework/cloud/data/module/deployer/cloudfoundry/ApplicationStatus.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.data.module.deployer.cloudfoundry;
+
+import java.util.HashMap;
+import java.util.Map;
+
+final class ApplicationStatus extends HashMap<String, ApplicationInstanceStatus> {
```

Comment:
Using inheritance here feels wrong.
Should either be replaced with
* dealing with just a dumb Map
* returning a wrapper (having `getId()`) around a plain Map at usage point (_e.g._ `GetApplicationsStatusResponse`)
* customizing Jackson ser/deser to a custom type (this `ApplicationStatus`) which would use delegation rather than inheritance